source/utilities.cc

```diff
@@ -938,12 +948,166 @@ namespace aspect
+                        {
+                          //The url Array contains a separate array for each column of data.
+                          // This will put each of these individual arrays into its own vector.
+                          urlArray->value(tmp);
+                          columns.push_back(tmp);
+
+                        }
+                      else
+                        {
+                          AssertThrow (false,
+                                       ExcMessage (std::string("Error when reading from url: ") + filename +
+                                                   " Maybe it was not of the correct type?"));
+                        }
+
+                    }
+                  else
+                    {
+                      AssertThrow (false,
+                                   ExcMessage (std::string("Error when reading from url: ") + filename +
+                                               " Maybe it was not of the correct type?"));
+                    }
+                }
+
+
+              //Add the POINTS data that is required and found at the top of the data file.
+              // The POINTS values are set as attributes inside a table.
+              // Loop through the Attribute table to locate the points values within
+              std::vector<std::string> points;
+              libdap::AttrTable *table;
+              for (libdap::AttrTable::Attr_iter i = das.var_begin(); i != das.var_end(); i++)
+                {
+                  table = das.get_table(i);
+                  if (table->get_attr("points") != "")
+                    points.push_back(table->get_attr("points"));
+                  else
+                    break;
+                }
+
+              //Append the gathered POINTS in the proper format:
+              // "# POINTS: <val1> <val2> <val3>"
+              urlString << "# POINTS:";
+              for (unsigned int i = 0; i < points.size(); i++)
+                {
+                  urlString << " " << points[i];
+                }
+              urlString << "\n";
+
+              //Add the values from the arrays into the stringstream. The values are passed in
+              // per row with a character return added at the end of each row.
+              // TODO: Add a check to make sure that each column is the same size before writing
+              //     to the stringstream
+              for (unsigned int i = 0; i < tmp.size(); i++)
+                {
+                  for (unsigned int j = 0; j < columns.size(); j++)
+                    {
+                      urlString << columns[j][i];
+                      urlString << " ";
+                    }
+                  urlString << "\n";
+                }
+
+              data_string = urlString.str();
+              filesize = data_string.size();
+            }
+
+          else
+            {
+              std::ifstream filestream(filename.c_str());
```

Comment:
is this just a copy&paste from the block below? I think you can reorder your ``ifdef`` and your ``if (starts with http)`` to avoid this copy.

Reply:
I'm not sure if I can/know how. The reason it's copied is in case the user has libdap turned on but is trying to read a local file. It's currently setup so that the `if (starts with http)` is triggered in the #else (meaning libdap is turned off) then it throws an error. Is there a way to setup the ifdef so that it can read a local file if libdap is on or off?